python/API_GW_EXECUTION_LOGGING_ENABLED/API_GW_EXECUTION_LOGGING_ENABLED_test.py

```diff
@@ -0,0 +1,286 @@
+import sys
+import unittest
+try:
+    from unittest.mock import MagicMock
+except ImportError:
+    from mock import MagicMock
+import json
+import botocore
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ApiGateway::RestApi'
+
+#############
+# Main Code #
+#############
+
+CONFIG_CLIENT_MOCK = MagicMock()
+STS_CLIENT_MOCK = MagicMock()
+
+class Boto3Mock():
+    @staticmethod
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('API_GW_EXECUTION_LOGGING_ENABLED')
+
+class ParameterTest(unittest.TestCase):
+
+    invoking_event_parameter_test_1 = json.dumps({'configurationItemDiff': {'changedProperties': {'Configuration.methodSettings.*/*.loggingLevel': {'previousValue': 'OFF', 'updatedValue': 'ERROR', 'changeType': 'UPDATE'}, 'Configuration.lastUpdatedDate': {'previousValue': 'Dec 12, 2018 9:01:20 AM', 'updatedValue': 'Apr 13, 2019 6:04:37 PM', 'changeType': 'UPDATE'}}, 'changeType': 'UPDATE'}, 'configurationItem': {'relatedEvents': [], 'relationships': [{'resourceId': 'arn:aws:apigateway:us-east-1::/restapis/abcd123fgh', 'resourceName': 'resource1', 'resourceType': 'AWS::ApiGateway::RestApi', 'name': 'Is contained in '}], 'configuration': {'deploymentId': '0r280l', 'stageName': 'test', 'cacheClusterEnabled': False, 'cacheClusterSize': '0.5', 'cacheClusterStatus': 'NOT_AVAILABLE', 'methodSettings': {'*/*': {'metricsEnabled': False, 'loggingLevel': 'ERROR', 'dataTraceEnabled': False, 'throttlingBurstLimit': 5000, 'throttlingRateLimit': 10000.0, 'cachingEnabled': False, 'cacheTtlInSeconds': 300, 'cacheDataEncrypted': False, 'requireAuthorizationForCacheControl': True, 'unauthorizedCacheControlHeaderStrategy': 'SUCCEED_WITH_RESPONSE_HEADER'}}, 'tracingEnabled': True, 'createdDate': 'Dec 4, 2018 10:09:21 AM', 'lastUpdatedDate': 'Apr 13, 2019 6:04:37 PM'}, 'supplementaryConfiguration': {}, 'tags': {}, 'configurationItemVersion': '1.3', 'configurationItemCaptureTime': '2019-04-13T18:04:37.523Z', 'configurationStateId': 1555178677523, 'awsAccountId': '123456789012', 'configurationItemStatus': 'OK', 'resourceType': 'AWS::ApiGateway::Stage', 'resourceId': 'arn:aws:apigateway:us-east-1::/restapis/abcd123fgh/stages/test', 'resourceName': 'test', 'ARN': 'arn:aws:apigateway:us-east-1::/restapis/abcd123fgh/stages/test', 'awsRegion': 'us-east-1', 'availabilityZone': 'Not Applicable', 'configurationStateMd5Hash': '', 'resourceCreationTime': '2018-12-04T10:09:21.128Z'}, 'notificationCreationTime': '2019-04-13T18:04:37.840Z', 'messageType': 'ConfigurationItemChangeNotification', 'recordVersion': '1.3'})
```

Comment:
Please remove the key/value not used by the rule. It is really hard to read like that.

Reply:
It might be worth to have a function that generate this json. There is a lot of repeat.